### PR TITLE
Tops and PuID

### PR DESCRIPTION
**Pull request by cros0400**
- Reverting to the NNLO / NLO top pt reweighing scheme
- Adding in pileup ID tight function 
- Noticed that we need to apply different requirements based on year. 2016pre and post are the same, 2017 and 2018 are the same but different from 2016.

### CODE REVIEW

**Comment by JHiltbrand** [comment]
And this top pt weight is setup to only apply to TTbar ?

**Comment by cros0400** [comment]
Adding the top suggestion in next commit.

Yes, see [here](https://github.com/StealthStop/Framework/blob/1302d6968639038f327c3da6a22f2a1b8d90c596/Framework/include/ScaleFactors.h#L661).

File: Framework/include/Jet.h [hunk 1]
```diff
@@ -1,13 +1,19 @@
 #ifndef JET_H
 #define JET_H
 
-inline bool passesPuIdMedium(float pt, float eta, float disc_value){
-     constexpr float cuts[4][4] = {
-        { 0.20,-0.56,-0.43,-0.38},
+inline bool passesPuIdMedium(float pt, float eta, float disc_value, std::string runYear){
+     constexpr float cuts[2][4][4] = {
```
Comment (cmadrid1) [inline Framework/include/Jet.h:5]: Is this object what is recommended? Can we make it a map or something?

Comment (cros0400) [reply Framework/include/Jet.h:5]: I'm sure we could. We just need a way to adjust our requirement based on eta and pt

Comment (cmadrid1) [reply Framework/include/Jet.h:5]: Ok at the very least it should be a member of the jet class. Looks like a global function at the moment.

Comment (cmadrid1) [reply Framework/include/Jet.h:5]: Also, can you make the array of arrays of arrays a member variable that is made by the constructor so we don't have to write it every event.

Comment (cros0400) [reply Framework/include/Jet.h:5]: Good call, I'll update to reflect

**Comment by cmadrid1** [comment]
+1

**Pull request merged**